core/src/main/java/org/elasticsearch/common/lucene/search/function/FieldValueFactorFunction.java

```diff
@@ -168,5 +168,13 @@ public double apply(double n) {
         public String toString() {
             return super.toString().toLowerCase(Locale.ROOT);
         }
+
+        public static Modifier fromString(String modifier) {
+            try {
+                return valueOf(modifier.toUpperCase(Locale.ROOT));
+            } catch (Throwable t) {
+                throw new IllegalArgumentException("Illegal modifier: " + modifier);
```

Comment:
Why not wrap the caught exception instead of swallowing the actual cause of the failure?